ads/dotandads.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {writeScript, dataHasAllFields} from '../src/3p';
+import {log} from '../src/log';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function dotandads(global, data) {
+  const mandatoryData = ["sp", "cid", "mpo", "mpt"];
+  if (dataHasAllFields(data, mandatoryData)) {
+    global.data = data;
+    writeScript(global, 'https://amp.ad.dotandad.com/dotandadsAmp.js');
+  } else {
+    log.error("DotAndAds", "missing mandatory fields");
+    global.context.noContentAvailable();
```

Comment:
This is technically fine, but unless you expect errors to happen in prod I would not hide the slot.